server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -723,6 +725,8 @@ protected Node(
                 threadPool
             );
 
+            remoteStoreStatsTrackerFactory = new RemoteStoreStatsTrackerFactory(clusterService, settings);
```

Comment:
Should we put the assignment behind the remote store feature flag?

Reply:
Tried this. However, Guice is encountering errors with this. This is because `IndicesClusterStateService` uses Guice to pull `RemoteStoreStatsTrackerFactory`. If remote store feature flag wasn't enabled, I had set the factory to null. However, it is not allowed to bind a null instance. Did not go ahead with the proposed change. Let me know if you have an alternate in mind.